src/kOS/Suffixed/Part/PartValueFactory.cs

```diff
@@ -6,45 +6,13 @@ namespace kOS.Suffixed.Part
 {
     public class PartValueFactory
     {
-        public static ListValue Construct(IEnumerable<global::Part> parts, SharedObjects shared)
-        {
-            var partList = parts.Select(part => Construct(part, shared)).ToList();
-            return ListValue.CreateList(partList);
-        }
+        public static ListValue Construct(IEnumerable<global::Part> parts, SharedObjects shared) =>
+            ListValue.CreateList(parts.Select(part => Construct(part, shared)));
 
-        public static ListValue<PartValue> ConstructGeneric(IEnumerable<global::Part> parts, SharedObjects shared)
-        {
-            var partList = parts.Select(part => Construct(part, shared)).ToList();
-            return ListValue<PartValue>.CreateList(partList);
-        }
+        public static ListValue<PartValue> ConstructGeneric(IEnumerable<global::Part> parts, SharedObjects shared) =>
+            ListValue<PartValue>.CreateList(parts.Select(part => Construct(part, shared)));
 
-        public static PartValue Construct(global::Part part, SharedObjects shared)
```

Comment:
This logic (`PartValueFactory.Construct`) was moved to `VesselTarget.Parts:ConstructPart`.
The whole file is now just small set of tiny helpers that could be moved to `VesselTarget.Parts` as well.

Reply:
Was there a reason for using `partial class` like this and defining one class in multiple files?  Did it solve a problem or was it just an aesthetic preference?  (Full disclosure, my aesthetic preference is against using ``partial class``.  When asking "where do I look to see how this class works?" I like getting one answer, not three.)

Reply:
Aesthetic/navigation. The file was just too big for effective navigation. I found myself scrolling a lot, splitting the window to see parts of the file at once... that is usually a sign to consider splitting it somehow, `partial class` is one possibility, creating helper classes is another (but often harder to execute).

I can merge it back to single file, if you don't like it. I separated it because it felt like three different parts that just happen to live inside single class, but has almost nothing to do with each other.

Reply:
I'd feel different about it if the syntax of ``partial class`` required some kind of header statement that mentioned what the rest of the class is (i.e. "this class is defined as the union of A, B, and C.  Now you can go look up A, B, and C in other files to get the full definition of this class".)  But partial class instead says, "The definition of this class is:  just grep for the places it's mentioned and paste them together in your head.  There is no actual place it's listed.".

That's what makes ``partial class`` a very ugly construct, in my opinion.

Reply:
I'll see what @hvacengi thinks about it.  Part of my resistance is that we use ``partial class`` nowhere else in the entire project.  So it's introducing a new feature to the source files.  But he may have a different opinion than me about it.

Reply:
Adding &lt;DependentUpon&gt; in .csproj can help, can do that, it joins all the file under one node in VS - like `MainForm.Designer.cs` under `MainForm.cs` - and the naming scheme is obvious: `Class.Part.cs`, no need to search for references, you have all the parts nicely together, just look at it as if it was just one file merged, kind of `#include` with additional header in the parts - using, namespace and class name.

Parts can be separated to its own class (e.g. `VesselParts`) or completely joined with `StageValue` (moving most of the stuff there, possibly even removing `PartFactory`). `VesselCache` can be created as `static class` to hold the instance cache and `private class Hooks` (`VesselTarget` only needs `IDisposable` reference to it, does not really need to know the real class, `InvalidateParts` can be made public, no harm in it, Hooks only need this one method) . That would reduce the amount of code in VesselTarget and separate it without introducing `partial class`.

P.S.: I miss `friend` from C++.

Reply:
Actually, you can have multiple source files (.cpp) for single class as well (some methods here, some there), the difference is that you need header file in C++, you don't in C#.

But general rule is: consider separate classes if you wish to split sources.